<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
  <head>
    <meta charset="utf-8" />
    <meta name="generator" content="pandoc" />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1.0, user-scalable=yes"
    />
    <title>README</title>
    <style type="text/css">
      code {
        white-space: pre-wrap;
      }
      span.smallcaps {
        font-variant: small-caps;
      }
      span.underline {
        text-decoration: underline;
      }
      div.column {
        display: inline-block;
        vertical-align: top;
        width: 50%;
      }
    </style>
  </head>
  <body>
    <h1 id="awesome-bioinformatics-awesome-url-check-toc">
      Awesome Bioinformatics
      <a href="https://github.com/sindresorhus/awesome"
        ><img
          src="https://cdn.rawgit.com/sindresorhus/awesome/d7305f38d29fed78fa85652e3a63e154dd8e8829/media/badge.svg"
          alt="Awesome"
      /></a>
      <img
        src="https://github.com/danielecook/Awesome-Bioinformatics/workflows/URL%20Check/badge.svg"
        alt="URL Check"
      />
      <img
        src="https://github.com/danielecook/Awesome-Bioinformatics/workflows/TOC/badge.svg"
        alt="TOC"
      />
    </h1>
    <blockquote>
      <p>
        Bioinformatics is an interdisciplinary field that develops methods and
        software tools for understanding biological data. —
        <a href="https://en.wikipedia.org/wiki/Bioinformatics">Wikipedia</a>
      </p>
    </blockquote>
    <p>
      A curated list of awesome Bioinformatics software, resources, and
      libraries. Mostly command line based, and free or open-source. Please feel
      free to <a href="CONTRIBUTING.md">contribute</a>!
    </p>
    <!-- START doctoc generated TOC please keep comment here to allow auto update -->
    <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
    <h2 id="table-of-contents">Table of Contents</h2>
    <ul>
      <li><a href="#package-suites">Package suites</a></li>
      <li><a href="#data-tools">Data Tools</a></li>
      <li>
        <a href="#data-processing">Data Processing</a>
        <ul>
          <li><a href="#command-line-utilities">Command Line Utilities</a></li>
        </ul>
      </li>
      <li>
        <a href="#next-generation-sequencing">Next Generation Sequencing</a>
        <ul>
          <li><a href="#workflow-managers">Workflow Managers</a></li>
          <li><a href="#pipelines">Pipelines</a></li>
          <li><a href="#sequence-processing">Sequence Processing</a></li>
          <li><a href="#data-analysis">Data Analysis</a></li>
          <li>
            <a href="#sequence-alignment">Sequence Alignment</a>
            <ul>
              <li><a href="#pairwise">Pairwise</a></li>
              <li>
                <a href="#multiple-sequence-alignment"
                  >Multiple Sequence Alignment</a
                >
              </li>
              <li><a href="#clustering">Clustering</a></li>
            </ul>
          </li>
          <li><a href="#quantification">Quantification</a></li>
          <li>
            <a href="#variant-calling">Variant Calling</a>
            <ul>
              <li>
                <a href="#structural-variant-callers"
                  >Structural variant callers</a
                >
              </li>
            </ul>
          </li>
          <li><a href="#bam-file-utilities">BAM File Utilities</a></li>
          <li><a href="#vcf-file-utilities">VCF File Utilities</a></li>
          <li><a href="#gff-bed-file-utilities">GFF BED File Utilities</a></li>
          <li><a href="#variant-simulation">Variant Simulation</a></li>
          <li>
            <a href="#variant-predictionannotation"
              >Variant Prediction/Annotation</a
            >
          </li>
          <li>
            <a href="#python-modules">Python Modules</a>
            <ul>
              <li><a href="#data">Data</a></li>
              <li><a href="#tools">Tools</a></li>
            </ul>
          </li>
          <li><a href="#assembly">Assembly</a></li>
          <li><a href="#annotation">Annotation</a></li>
        </ul>
      </li>
      <li>
        <a href="#long-read-sequencing">Long-read sequencing</a>
        <ul>
          <li><a href="#long-read-assembly">Long-read Assembly</a></li>
        </ul>
      </li>
      <li>
        <a href="#visualization">Visualization</a>
        <ul>
          <li>
            <a href="#genome-browsers--gene-diagrams"
              >Genome Browsers / Gene Diagrams</a
            >
          </li>
          <li><a href="#circos-related">Circos Related</a></li>
        </ul>
      </li>
      <li><a href="#database-access">Database Access</a></li>
      <li>
        <a href="#resources">Resources</a>
        <ul>
          <li>
            <a href="#becoming-a-bioinformatician"
              >Becoming a Bioinformatician</a
            >
          </li>
          <li>
            <a href="#bioinformatics-on-github">Bioinformatics on GitHub</a>
          </li>
          <li><a href="#sequencing">Sequencing</a></li>
          <li><a href="#rna-seq">RNA-Seq</a></li>
          <li><a href="#chip-seq">ChIP-Seq</a></li>
          <li>
            <a href="#youtube-channels-and-playlists"
              >YouTube Channels and Playlists</a
            >
          </li>
          <li><a href="#blogs">Blogs</a></li>
          <li><a href="#miscellaneous">Miscellaneous</a></li>
        </ul>
      </li>
      <li><a href="#online-networking-groups">Online networking groups</a></li>
      <li><a href="#license">License</a></li>
    </ul>
    <!-- END doctoc generated TOC please keep comment here to allow auto update -->
    <hr />
    <h2 id="package-suites">Package suites</h2>
    <p>
      Package suites gather software packages and installation tools for
      specific languages or platforms. We have some for bioinformatics software.
    </p>
    <ul>
      <li>
        <p>
          <strong
            ><a href="https://github.com/Bioconductor">Bioconductor</a></strong
          >
          - A plethora of tools for analysis and comprehension of
          high-throughput genomic data, including 1500+ software packages. [
          <a href="https://link.springer.com/article/10.1186/gb-2004-5-10-r80"
            >paper-2004</a
          >
          | <a href="https://www.bioconductor.org">web</a> ]
        </p>
      </li>
      <li>
        <p>
          <strong
            ><a href="https://github.com/biopython/biopython"
              >Biopython</a
            ></strong
          >
          - Freely available tools for biological computing in Python, with
          included cookbook, packaging and thorough documentation. Part of the
          <a href="http://open-bio.org/">Open Bioinformatics Foundation</a>.
          Contains the very useful
          <a href="https://biopython.org/DIST/docs/api/Bio.Entrez-module.html"
            >Entrez</a
          >
          package for API access to the NCBI databases. [
          <a href="https://pubmed.ncbi.nlm.nih.gov/19304878">paper-2009</a> |
          <a href="https://biopython.org">web</a> ]
        </p>
      </li>
      <li>
        <p>
          <strong><a href="https://github.com/bioconda">Bioconda</a></strong> -
          A channel for the
          <a href="http://conda.pydata.org/docs/intro.html"
            >conda package manager</a
          >
          specializing in bioinformatics software. Includes a repository with
          3000+ ready-to-install (with <code>conda install</code>)
          bioinformatics packages. [
          <a href="https://pubmed.ncbi.nlm.nih.gov/29967506">paper-2018</a> |
          <a href="https://bioconda.github.io">web</a> ]
        </p>
      </li>
      <li>
        <strong><a href="https://github.com/BioJulia">BioJulia</a></strong> -
        Bioinformatics and computational biology infastructure for the Julia
        programming language. [ <a href="https://biojulia.net">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/rust-bio/rust-bio">Rust-Bio</a></strong
        >
        - Rust implementations of algorithms and data structures useful for
        bioinformatics. [
        <a
          href="http://bioinformatics.oxfordjournals.org/content/early/2015/10/06/bioinformatics.btv573.short?rss=1"
          >paper-2016</a
        >
        ]
      </li>
      <li>
        <strong><a href="https://github.com/seqan/seqan3">SeqAn</a></strong> -
        The modern C++ library for sequence analysis.
      </li>
      <li>
        <strong
          ><a href="https://github.com/TimothyStiles/poly"
            >(Poly)merase</a
          ></strong
        >
        - A Go library and command line utility for engineering organisms.
      </li>
      <li>
        <p>
          <strong
            ><a href="https://github.com/biocaml/biocaml">Biocaml</a></strong
          >
          - Biocaml aims to be a high-performance user-friendly library for
          Bioinformatics.
        </p>
      </li>
    </ul>
    <h2 id="data-tools">Data Tools</h2>
    <ul>
      <li>
        <strong><a href="https://github.com/gogetdata/ggd-cli">GGD</a></strong>
        - Go Get Data; A command line interface for obtaining genomic data. [
        <a href="https://gogetdata.github.io">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/ewels/sra-explorer"
            >SRA-Explorer</a
          ></strong
        >
        - Easily get SRA download links and other information. [
        <a href="https://sra-explorer.info">web</a> ]
      </li>
    </ul>
    <h2 id="data-processing">Data Processing</h2>
    <h3 id="command-line-utilities">Command Line Utilities</h3>
    <ul>
      <li>
        <strong
          ><a href="https://github.com/stephenturner/oneliners"
            >Bioinformatics One Liners</a
          ></strong
        >
        - Git repo of useful single line commands.
      </li>
      <li>
        <strong
          ><a href="https://github.com/bionode/bionode">BioNode</a></strong
        >
        - Modular and universal bioinformatics, Bionode provides pipeable UNIX
        command line tools and JavaScript APIs for bioinformatics analysis
        workflows. [ <a href="http://bionode.io">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/bioSyntax/bioSyntax"
            >bioSyntax</a
          ></strong
        >
        - Syntax Highlighting for Computational Biology file formats (SAM, VCF,
        GTF, FASTA, PDB, etc…) in vim/less/gedit/sublime. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/30134911">paper-2018</a> |
        <a href="http://www.bioSyntax.org">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/wireservice/csvkit">CSVKit</a></strong
        >
        - Utilities for working with CSV/Tab-delimited files. [
        <a href="https://csvkit.readthedocs.io/en/latest">web</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/shenwei356/csvtk">csvtk</a></strong>
        - Another cross-platform, efficient, practical and pretty CSV/TSV
        toolkit. [ <a href="https://bioinf.shenwei.me/csvtk">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://git.savannah.gnu.org/gitweb/?p=datamash.git"
            >datamash</a
          ></strong
        >
        - Data transformations and statistics. [
        <a href="http://www.gnu.org/software/datamash">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/shenwei356/easy_qsub"
            >easy_qsub</a
          ></strong
        >
        - Easily submitting PBS jobs with script template. Multiple input files
        supported.
      </li>
      <li>
        <strong>GNU Parallel</strong> - General parallelizer that runs jobs in
        parallel on a single multi-core machine.
        <a href="https://www.biostars.org/p/63816/">Here</a> are some example
        scripts using GNU Parallel. [
        <a href="http://www.gnu.org/software/parallel">web</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/arq5x/grabix">grabix</a></strong> -
        A wee tool for random access into BGZF files.
      </li>
      <li>
        <strong><a href="https://github.com/brentp/gsort">gsort</a></strong> -
        Sort genomic files according to a specified order.
      </li>
      <li>
        <strong><a href="https://github.com/samtools/tabix">tabix</a></strong> -
        Table file index. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/21208982">paper-2011</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/wormtable/wormtable"
            >wormtable</a
          ></strong
        >
        - Write-once-read-many table for large datasets.
      </li>
      <li>
        <strong
          ><a href="https://github.com/mattgodbolt/zindex">zindex</a></strong
        >
        - Create an index on a compressed text file.
      </li>
    </ul>
    <h2 id="next-generation-sequencing">Next Generation Sequencing</h2>
    <h3 id="workflow-managers">Workflow Managers</h3>
    <ul>
      <li>
        <strong
          ><a href="https://github.com/pcingola/BigDataScript"
            >BigDataScript</a
          ></strong
        >
        - A cross-system scripting language for working with big data pipelines
        in computer systems of different sizes and capabilities. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/25189778">paper-2014</a> |
        <a href="https://pcingola.github.io/BigDataScript">web</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/ssadedin/bpipe">Bpipe</a></strong> -
        A small language for defining pipeline stages and linking them together
        to make pipelines. [ <a href="http://docs.bpipe.org">web</a> ]
      </li>
      <li>
        <strong
          ><a
            href="https://github.com/common-workflow-language/common-workflow-language"
            >Common Workflow Language</a
          ></strong
        >
        - a specification for describing analysis workflows and tools that are
        portable and scalable across a variety of software and hardware
        environments, from workstations to cluster, cloud, and high performance
        computing (HPC) environments. [
        <a href="http://www.commonwl.org">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/broadinstitute/cromwell"
            >Cromwell</a
          ></strong
        >
        - A Workflow Management System geared towards scientific workflows. [
        <a href="https://cromwell.readthedocs.io">web</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/galaxyproject">Galaxy</a></strong> -
        a popular open-source, web-based platform for data intensive biomedical
        research. Has several features, from data analysis to workflow
        management to visualization tools. [
        <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6030816"
          >paper-2018</a
        >
        | <a href="https://galaxyproject.org">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/nextflow-io/nextflow">Nextflow</a>
          (recommended)</strong
        >
        - A fluent DSL modelled around the UNIX pipe concept, that simplifies
        writing parallel and scalable pipelines in a portable manner. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/29412134">paper-2018</a> |
        <a href="http://nextflow.io">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/cgat-developers/ruffus"
            >Ruffus</a
          ></strong
        >
        - Computation Pipeline library for python widely used in science and
        bioinformatics. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/20847218">paper-2010</a> |
        <a href="http://www.ruffus.org.uk">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/SeqWare/seqware">SeqWare</a></strong
        >
        - Hadoop Oozie-based workflow system focused on genomics data analysis
        in cloud environments. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/21210981">paper-2010</a> |
        <a href="https://seqware.github.io">web</a> ]
      </li>
      <li>
        <strong><a href="https://bitbucket.org/snakemake">Snakemake</a></strong>
        - A workflow management system in Python that aims to reduce the
        complexity of creating workflows by providing a fast and comfortable
        execution environment. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/29788404">paper-2018</a> |
        <a href="https://snakemake.readthedocs.io">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/broadinstitute/wdl"
            >Workflow Descriptor Language</a
          ></strong
        >
        - Workflow standard developed by the Broad. [
        <a href="https://software.broadinstitute.org/wdl">web</a> ]
      </li>
    </ul>
    <h3 id="pipelines">Pipelines</h3>
    <ul>
      <li>
        <strong
          ><a href="https://github.com/pditommaso/awesome-pipeline"
            >Awesome-Pipeline</a
          ></strong
        >
        - A list of pipeline resources.
      </li>
      <li>
        <strong
          ><a href="https://github.com/bactopia/bactopia/">Bactopia</a></strong
        >
        - A flexible pipeline, built with Nextflow, for the complete analysis of
        bacterial genomes. [ <a href="https://bactopia.github.io/">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/fmalmeida/bacannot">Bacannot</a></strong
        >
        - A generic but comprehensive bacterial annotation pipeline, built with
        Nextflow, with nice graphical options for investigating results. [
        <a href="https://bacannot.readthedocs.io/en/latest/?badge=latest"
          >web</a
        >
        ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/chapmanb/bcbio-nextgen"
            >bcbio-nextgen</a
          ></strong
        >
        - Batteries included genomic analysis pipeline for variant and RNA-Seq
        analysis, structural variant calling, annotation, and prediction. [
        <a href="https://bcbio-nextgen.readthedocs.io">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/pentalpha/r-peridot"
            >R-Peridot</a
          ></strong
        >
        - Customizable pipeline for differential expression analysis with an
        intuitive GUI. [
        <a href="http://www.bioinformatics-brazil.org/r-peridot">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/fmalmeida/ngs-preprocess"
            >ngs-preprocess</a
          ></strong
        >
        - A pipeline for preprocessing short and long sequencing reads, built
        with Nextflow. [
        <a href="https://ngs-preprocess.readthedocs.io/en/latest/?badge=latest"
          >web</a
        >
        ]
      </li>
    </ul>
    <h3 id="sequence-processing">Sequence Processing</h3>
    <p>
      Sequence Processing includes tasks such as demultiplexing raw read data,
      and trimming low quality bases.
    </p>
    <ul>
      <li>
        <strong
          ><a href="https://github.com/OpenGene/AfterQC">AfterQC</a></strong
        >
        - Automatic Filtering, Trimming, Error Removing and Quality Control for
        fastq data. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/28361673">paper-2017</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/s-andrews/FastQC">FastQC</a></strong
        >
        - A quality control tool for high throughput sequence data. [
        <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc"
          >web</a
        >
        ]
      </li>
      <li>
        <strong><a href="https://github.com/mdshw5/fastqp">Fastqp</a></strong> -
        FASTQ and SAM quality control using Python.
      </li>
      <li>
        <strong
          ><a href="https://github.com/agordon/fastx_toolkit"
            >Fastx Tookit</a
          ></strong
        >
        - FASTQ/A short-reads pre-processing tools: Demultiplexing, trimming,
        clipping, quality filtering, and masking utilities. [
        <a href="http://hannonlab.cshl.edu/fastx_toolkit">web</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/ewels/MultiQC">MultiQC</a></strong>
        - Aggregate results from bioinformatics analyses across many samples
        into a single report. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/27312411">paper-2016</a> |
        <a href="http://multiqc.info">web</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/telatin/seqfu2">SeqFu</a></strong> -
        Sequence manipulation toolkit for FASTA/FASTQ files written in Nim. [
        <a href="https://www.mdpi.com/2306-5354/8/5/59">paper-2021</a> |
        <a href="https://telatin.github.io/seqfu2/">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/shenwei356/seqkit">SeqKit</a></strong
        >
        - A cross-platform and ultrafast toolkit for FASTA/Q file manipulation
        in Golang. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/27706213">paper-2016</a> |
        <a href="https://bioinf.shenwei.me/seqkit">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/fhcrc/seqmagick">seqmagick</a></strong
        >
        - file format conversion in Biopython in a convenient way. [
        <a href="http://seqmagick.readthedocs.io">web</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/lh3/seqtk">Seqtk</a></strong> -
        Toolkit for processing sequences in FASTA/Q formats.
      </li>
      <li>
        <strong
          ><a href="https://github.com/incertae-sedis/smof">smof</a></strong
        >
        - UNIX-style FASTA manipulation tools.
      </li>
    </ul>
    <h3 id="data-analysis">Data Analysis</h3>
    <p>
      The following items allow for scalable genomic analysis by introducing
      specialized databases.
    </p>
    <ul>
      <li>
        <strong><a href="https://github.com/hail-is/hail">Hail</a></strong> -
        Scalable genomic analysis.
      </li>
      <li>
        <strong
          ><a href="https://github.com/dnanexus-rnd/GLnexus">GLNexus</a></strong
        >
        - Scalable gVCF merging and joint variant calling for population
        sequencing projects. [
        <a href="https://www.biorxiv.org/content/10.1101/343970v1.abstract"
          >paper-2018</a
        >
        ]
      </li>
    </ul>
    <h3 id="sequence-alignment">Sequence Alignment</h3>
    <h4 id="pairwise">Pairwise</h4>
    <ul>
      <li>
        <strong
          ><a href="https://github.com/BenLangmead/bowtie2">Bowtie 2</a></strong
        >
        - An ultrafast and memory-efficient tool for aligning sequencing reads
        to long reference sequences. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/22388286">paper-2012</a> |
        <a href="http://bowtie-bio.sourceforge.net/bowtie2">web</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/lh3/bwa">BWA</a></strong> -
        Burrow-Wheeler Aligner for pairwise alignment between DNA sequences.
      </li>
      <li>
        <strong><a href="https://github.com/smarco/WFA">WFA</a></strong> - the
        wavefront alignment algorithm (WFA) which expoit sequence similarity to
        speed up alignment [
        <a
          href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btaa777/5904262"
          >paper-2020</a
        >
        ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/jeffdaily/parasail">Parasail</a></strong
        >
        - SIMD C library for global, semi-global, and local pairwise sequence
        alignments [
        <a
          href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-0930-z"
          >paper-2016</a
        >
        ]
      </li>
      <li>
        <strong><a href="https://github.com/mummer4/mummer">MUMmer</a></strong>
        - A system for rapidly aligning entire genomes, whether in complete or
        draft form. [
        <a href="http://mummer.sourceforge.net/MUMmer.pdf">paper-1999</a> |
        <a href="http://mummer.sourceforge.net/MUMmer2.pdf">paper-2002</a> |
        <a href="http://mummer.sourceforge.net/MUMmer3.pdf">paper-2004</a> |
        <a href="http://mummer.sourceforge.net">web</a> ]
      </li>
    </ul>
    <h4 id="multiple-sequence-alignment">Multiple Sequence Alignment</h4>
    <ul>
      <li>
        <strong><a href="https://github.com/ljdursi/poapy">POA</a></strong> -
        Partial-Order Alignment for fast alignment and consensus of multiple
        homologous sequences. [
        <a
          href="https://academic.oup.com/bioinformatics/article/18/3/452/236691"
          >paper-2002</a
        >
        ]
      </li>
    </ul>
    <h4 id="clustering">Clustering</h4>
    <ul>
      <li>
        <strong
          ><a href="https://github.com/soedinglab/MMseqs2">MMseqs2</a></strong
        >
        - Ultra-fast, sensitive search and clustering suite for protein and
        nucleotide sequence sets. [
        <a href="https://www.nature.com/articles/nbt.3988">paper-2017</a> |
        <a href="https://www.nature.com/articles/s41467-018-04964-5"
          >paper-2018</a
        >
        ]
      </li>
    </ul>
    <h3 id="quantification">Quantification</h3>
    <ul>
      <li>
        <strong
          ><a href="https://github.com/cole-trapnell-lab/cufflinks"
            >Cufflinks</a
          ></strong
        >
        - Cufflinks assembles transcripts, estimates their abundances, and tests
        for differential expression and regulation in RNA-Seq samples. [
        <a href="https://www.nature.com/articles/nbt.1621">paper-2010</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/deweylab/RSEM">RSEM</a></strong> - A
        software package for estimating gene and isoform expression levels from
        RNA-Seq data. [
        <a
          href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-323"
          >paper-2011</a
        >
        | <a href="http://deweylab.github.io/RSEM/">web</a> ]
      </li>
    </ul>
    <h3 id="variant-calling">Variant Calling</h3>
    <ul>
      <li>
        <strong
          ><a href="https://github.com/google/deepvariant"
            >DeepVariant</a
          ></strong
        >
        - Deep learning-based variant caller [
        <a href="https://rdcu.be/7Dhl">paper-2018</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/ekg/freebayes">freebayes</a></strong
        >
        - Bayesian haplotype-based polymorphism discovery and genotyping. [
        <a href="http://arxiv.org/abs/1207.3907">web</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/broadgsa/gatk">GATK</a></strong> -
        Variant Discovery in High-Throughput Sequencing Data. [
        <a href="https://software.broadinstitute.org/gatk">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/luntergroup/octopus">Octopus</a></strong
        >
        - A polymorphic bayesian genotyping model with wide applicability. [
        <a href="https://www.nature.com/articles/s41587-021-00861-3"
          >paper-2021</a
        >
        ]
      </li>
      <li>
        <p>
          <strong
            ><a href="https://github.com/samtools/bcftools">bcftools</a></strong
          >
          - samtools/bcftools are a suite of tools for manipulating NGS data and
          can be used to call variants. [
          <a href="https://pubmed.ncbi.nlm.nih.gov/19505943">paper-2009</a> |
          <a href="http://htslib.org">web</a> ] #### Structural variant callers
        </p>
      </li>
      <li>
        <strong><a href="https://github.com/dellytools/delly">Delly</a></strong>
        - Structural variant discovery by integrated paired-end and split-read
        analysis. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/22962449">paper-2012</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/arq5x/lumpy-sv">lumpy</a></strong> -
        lumpy: a general probabilistic framework for structural variant
        discovery. [
        <a href="https://link.springer.com/article/10.1186/gb-2014-15-6-r84"
          >paper-2014</a
        >
        ]
      </li>
      <li>
        <strong><a href="https://github.com/Illumina/manta">manta</a></strong> -
        Structural variant and indel caller for mapped sequencing data. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/26647377">paper-2015</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/PapenfussLab/gridss">gridss</a></strong
        >
        - GRIDSS: the Genomic Rearrangement IDentification Software Suite. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/29097403">paper-2017</a> ]
      </li>
      <li>
        <p>
          <strong><a href="https://github.com/brentp/smoove">smoove</a></strong>
          - structural variant calling and genotyping with existing tools,
          but,smoothly.
        </p>
      </li>
    </ul>
    <h3 id="bam-file-utilities">BAM File Utilities</h3>
    <ul>
      <li>
        <strong
          ><a href="https://github.com/pezmaster31/bamtools"
            >Bamtools</a
          ></strong
        >
        - Collection of tools for working with BAM files. [
        <a
          href="https://academic.oup.com/bioinformatics/article/27/12/1691/255399"
          >paper-2011</a
        >
        ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/AndersenLab/bam-toolbox"
            >bam toolbox</a
          ></strong
        >
        MtDNA:Nuclear Coverage; BAM Toolbox can output the ratio of
        MtDNA:nuclear coverage, a proxy for mitochondrial content.
      </li>
      <li>
        <strong
          ><a href="https://github.com/DarwinAwardWinner/mergesam"
            >mergesam</a
          ></strong
        >
        - Automate common SAM &amp; BAM conversions.
      </li>
      <li>
        <strong
          ><a href="https://github.com/brentp/mosdepth">mosdepth</a></strong
        >
        - fast BAM/CRAM depth calculation for WGS, exome, or targeted
        sequencing. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/29096012/">paper-2017</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/TimoLassmann/samstat">SAMstat</a></strong
        >
        - Displaying sequence statistics for next-generation sequencing. [
        <a
          href="https://academic.oup.com/bioinformatics/article/27/1/130/201972"
          >paper-2010</a
        >
        | <a href="http://samstat.sourceforge.net">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/brentp/mosdepth">Somalier</a></strong
        >
        - Fast sample-swap and relatedness checks on BAMs/CRAMs/VCFs/GVCFs. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/32664994">paper-2020</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/zd1/telseq">Telseq</a></strong> -
        Telseq is a tool for estimating telomere length from whole genome
        sequence data. [
        <a href="https://academic.oup.com/nar/article/42/9/e75/1249448"
          >paper-2014</a
        >
        ]
      </li>
    </ul>
    <h3 id="vcf-file-utilities">VCF File Utilities</h3>
    <ul>
      <li>
        <strong
          ><a href="https://github.com/samtools/bcftools">bcftools</a></strong
        >
        - Set of tools for manipulating VCF files. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/26826718">paper-2016</a> |
        <a href="https://pubmed.ncbi.nlm.nih.gov/28205675">paper-2017</a> |
        <a href="http://samtools.github.io/bcftools">web</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/brentp/vcfanno">vcfanno</a></strong>
        - Annotate a VCF with other VCFs/BEDs/tabixed files. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/27250555">paper-2016</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/vcflib/vcflib">vcflib</a></strong> -
        A C++ library for parsing and manipulating VCF files.
      </li>
      <li>
        <strong
          ><a href="https://github.com/vcftools/vcftools">vcftools</a></strong
        >
        - VCF manipulation and statistics (e.g. linkage disequilibrium, allele
        frequency, Fst). [
        <a href="https://pubmed.ncbi.nlm.nih.gov/21653522">paper-2011</a> ]
      </li>
    </ul>
    <h3 id="gff-bed-file-utilities">GFF BED File Utilities</h3>
    <ul>
      <li>
        <strong><a href="https://github.com/NBISweden/AGAT">AGAT</a></strong> -
        Suite of tools to handle gene annotations in any GTF/GFF format. [
        <a href="https://agat.readthedocs.io/en/latest/?badge=latest">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/daler/gffutils">gffutils</a></strong
        >
        - GFF and GTF file manipulation and interconversion. [
        <a href="http://daler.github.io/gffutils">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://bedops.readthedocs.io/en/latest/index.html"
            >BEDOPS</a
          ></strong
        >
        - The fast, highly scalable and easily-parallelizable genome analysis
        toolkit. [
        <a
          href="https://academic.oup.com/bioinformatics/article/28/14/1919/218826"
          >paper-2012</a
        >
        ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/arq5x/bedtools2">Bedtools2</a></strong
        >
        - A Swiss Army knife for genome arithmetic. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/20110278">paper-2010</a> |
        <a href="https://pubmed.ncbi.nlm.nih.gov/25199790">paper-2014</a> |
        <a href="https://bedtools.readthedocs.io">web</a> ]
      </li>
    </ul>
    <h3 id="variant-simulation">Variant Simulation</h3>
    <ul>
      <li>
        <strong
          ><a href="https://github.com/adamewing/bamsurgeon"
            >Bam Surgeon</a
          ></strong
        >
        - Tools for adding mutations to existing <code>.bam</code> files, used
        for testing mutation callers. [
        <a
          href="https://popmodels.cancercontrol.cancer.gov/gsr/packages/bamsurgeon"
          >web</a
        >
        ]
      </li>
      <li>
        <strong><a href="https://github.com/lh3/wgsim">wgsim</a></strong> -
        <strong>Comes with samtools!</strong> - Reads simulator. [
        <a href="https://popmodels.cancercontrol.cancer.gov/gsr/packages/wgsim"
          >web</a
        >
        ]
      </li>
    </ul>
    <h3 id="variant-predictionannotation">Variant Prediction/Annotation</h3>
    <ul>
      <li>
        <strong><a href="https://github.com/teamdfir/sift">SIFT</a></strong> -
        Predicts whether an amino acid substitution affects protein function. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/12824425">paper-2003</a> |
        <a href="http://sift.jcvi.org">web</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/pcingola/SnpEff">SnpEff</a></strong>
        - Genetic variant annotation and effect prediction toolbox. [
        <a href="https://www.tandfonline.com/doi/full/10.4161/fly.19695"
          >paper-2012</a
        >
        | <a href="https://pcingola.github.io/SnpEff">web</a> ]
      </li>
    </ul>
    <h3 id="python-modules">Python Modules</h3>
    <h4 id="data">Data</h4>
    <ul>
      <li>
        <strong><a href="https://github.com/brentp/cruzdb">cruzdb</a></strong> -
        Pythonic access to the UCSC Genome database. [
        <a
          href="https://academic.oup.com/bioinformatics/article/29/23/3003/248468"
          >paper-2013</a
        >
        ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/openvax/pyensembl">pyensembl</a></strong
        >
        - Pythonic Access to the Ensembl database. [
        <a href="https://pyensembl.readthedocs.io/en/latest/pyensembl.html"
          >web</a
        >
        ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/cokelaer/bioservices"
            >bioservices</a
          ></strong
        >
        - Access to Biological Web Services from Python. [
        <a
          href="https://academic.oup.com/bioinformatics/article/29/24/3241/194040"
          >paper-2013</a
        >
        | <a href="http://bioservices.readthedocs.io">web</a> ]
      </li>
    </ul>
    <h4 id="tools">Tools</h4>
    <ul>
      <li>
        <strong><a href="https://github.com/arq5x/cyvcf">cyvcf</a></strong> - A
        port of <a href="https://github.com/jamescasbon/PyVCF">pyVCF</a> using
        Cython for speed.
      </li>
      <li>
        <strong><a href="https://github.com/brentp/cyvcf2">cyvcf2</a></strong> -
        Cython + HTSlib == fast VCF parsing; even faster parsing than pyVCF. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/28165109">paper-2017</a> |
        <a href="https://brentp.github.io/cyvcf2">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/daler/pybedtools">pyBedTools</a></strong
        >
        - Python wrapper for
        <a href="https://github.com/arq5x/bedtools">bedtools</a>. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/21949271">paper-2011</a> |
        <a href="http://daler.github.io/pybedtools">web</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/mdshw5/pyfaidx">pyfaidx</a></strong>
        - Pythonic access to FASTA files.
      </li>
      <li>
        <strong
          ><a href="https://github.com/pysam-developers/pysam">pysam</a></strong
        >
        - Python wrapper for
        <a href="https://github.com/samtools/samtools">samtools</a>. [
        <a href="https://pysam.readthedocs.io/en/latest/api.html">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/jamescasbon/PyVCF">pyVCF</a></strong
        >
        - A VCF Parser for Python. [
        <a href="http://pyvcf.readthedocs.org/en/latest/index.html">web</a> ]
      </li>
    </ul>
    <h3 id="assembly">Assembly</h3>
    <ul>
      <li>
        <strong><a href="https://github.com/ablab/spades">SPAdes</a></strong> -
        SPAdes (St. Petersburg genome assembler) is an assembly toolkit
        containing various assembly pipelines and the de-facto standard for
        prokaryotic genome assemblies.
      </li>
      <li>
        <strong><a href="https://github.com/ncbi/SKESA">SKESA</a></strong> -
        SKESA is a de-novo sequence read assembler for microbial genomes. It
        uses conservative heuristics and is designed to create breaks at repeat
        regions in the genome. This leads to excellent sequence quality without
        significantly compromising contiguity.
      </li>
    </ul>
    <h3 id="annotation">Annotation</h3>
    <ul>
      <li>
        <strong><a href="https://github.com/tseemann/prokka">Prokka</a></strong>
        - Prokka: rapid prokaryotic genome annotation. Prokka is one of the most
        cited annotation command line tools for microbial genome annotations.
      </li>
      <li>
        <strong
          ><a href="https://github.com/oschwengers/bakta">Bakta</a></strong
        >
        - Bakta is a tool for the rapid &amp; standardized annotation of
        bacterial genomes &amp; plasmids. It provides dbxref-rich and
        sORF-including annotations in machine-readable JSON &amp; bioinformatics
        standard file formats for automatic downstream analysis.
      </li>
    </ul>
    <h2 id="long-read-sequencing">Long-read sequencing</h2>
    <h3 id="long-read-assembly">Long-read Assembly</h3>
    <ul>
      <li>
        <strong><a href="https://github.com/marbl/canu">canu</a></strong> - A
        single molecule sequence assembler for genomes large and small.
      </li>
      <li>
        <strong><a href="https://github.com/fenderglass/Flye">flye</a></strong>
        - De novo assembler for single molecule sequencing reads using repeat
        graphs.
      </li>
      <li>
        <strong
          ><a href="https://github.com/chhylp123/hifiasm">hifiasm</a></strong
        >
        - A haplotype-resolved assembler for accurate Hifi reads.
      </li>
      <li>
        <strong><a href="https://github.com/ruanjue/wtdbg2">wtdbg2</a></strong>
        - A fuzzy Bruijn graph approach to long noisy reads assembly
      </li>
    </ul>
    <h2 id="visualization">Visualization</h2>
    <h3 id="genome-browsers-gene-diagrams">Genome Browsers / Gene Diagrams</h3>
    <p>
      The following tools can be used to visualize genomic data or for
      constructing customized visualizations of genomic data including sequence
      data from DNA-Seq, RNA-Seq, and ChIP-Seq, variants, and more.
    </p>
    <ul>
      <li>
        <strong
          ><a href="https://github.com/Lab41/squiggle">Squiggle</a></strong
        >
        - Easy-to-use DNA sequence visualization tool that turns FASTA files
        into browser-based visualizations. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/30247632">paper-2018</a> |
        <a href="https://squiggle.readthedocs.io/en/latest/">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/dasmoth/dalliance"
            >biodalliance</a
          ></strong
        >
        - Embeddable genome viewer. Integration data from a wide variety of
        sources, and can load data directly from popular genomics file formats
        including bigWig, BAM, and VCF. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/21252075">paper-2011</a> |
        <a href="http://www.biodalliance.org">web</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/biojs/biojs">BioJS</a></strong> -
        BioJS is a library of over hundred JavaScript components enabling you to
        visualize and process data using current web technologies. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/25075290/">paper-2014</a> |
        <a href="http://biojs.net/">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/jonathancrabtree/Circleator"
            >Circleator</a
          ></strong
        >
        - Flexible circular visualization of genome-associated data with BioPerl
        and SVG. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/25075113">paper-2014</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/drio/dnaism">DNAism</a></strong> -
        Horizon chart D3-based JavaScript library for DNA data. [
        <a
          href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-0891-2"
          >paper-2016</a
        >
        | <a href="http://drio.github.io/dnaism/">web</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/igvteam/igv">IGV js</a></strong> -
        Java-based browser. Fast, efficient, scalable visualization tool for
        genomics data and annotations. Handles a large variety of formats. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/31099383">paper-2019</a> |
        <a href="https://software.broadinstitute.org/software/igv">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/lairdm/islandplot"
            >Island Plot</a
          ></strong
        >
        - D3 JavaScript based genome viewer. Constructs SVGs. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/25916842/">paper-2015</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/GMOD/jbrowse">JBrowse</a></strong> -
        JavaScript genome browser that is highly customizable via plugins and
        track customizations. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/27072794">paper-2016</a> |
        <a href="http://jbrowse.org/">web</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/chgibb/PHAT">PHAT</a></strong> -
        Point and click, cross platform suite for analysing and visualizing
        next-generation sequencing datasets. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/30561651">paper-2018</a> |
        <a href="https://chgibb.github.io/PHATDocs">web</a> ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/hammerlab/pileup.js"
            >pileup.js</a
          ></strong
        >
        - JavaScript library that can be used to generate interactive and highly
        customizable web-based genome browsers. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/27153605">paper-2016</a> ]
      </li>
      <li>
        <strong><a href="https://github.com/chmille4/Scribl">scribl</a></strong>
        - JavaScript library for drawing canvas-based gene diagrams. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/23172864">paper-2012</a> |
        <a href="http://chmille4.github.io/Scribl">web</a> ]
      </li>
      <li>
        <strong>Lucid Align</strong> - A modern sequence alignment viewer. [
        <a href="https://lucidalign.com">web</a> ]
      </li>
    </ul>
    <h3 id="circos-related">Circos Related</h3>
    <ul>
      <li>
        <strong
          ><a href="https://github.com/vigsterkr/circos">Circos</a></strong
        >
        - Perl package for circular plots, which are well suited for genomic
        rearrangements. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/19541911">paper-2009</a> |
        <a href="http://circos.ca">web</a> ]
      </li>
      <li>
        <strong>ClicO FS</strong> - An interactive web-based service of Circos.
        [ <a href="https://pubmed.ncbi.nlm.nih.gov/26227146">paper-2015</a> ]
      </li>
      <li>
        <strong>OmicCircos</strong> - R package for circular plots for omics
        data. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/24526832">paper-2014</a> |
        <a
          href="http://www.bioconductor.org/packages/release/bioc/html/OmicCircos.html"
          >web</a
        >
        ]
      </li>
      <li>
        <strong>J-Circos</strong> - A Java application for doing interactive
        work with circos plots. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/25540184">paper-2014</a> |
        <a
          href="http://www.australianprostatecentre.org/research/software/jcircos"
          >web</a
        >
        ]
      </li>
      <li>
        <strong
          ><a href="https://bitbucket.org/henryhzhang/rcircos/src/master/"
            >rCircos</a
          ></strong
        >
        - R package for circular plots. [
        <a href="https://pubmed.ncbi.nlm.nih.gov/23937229">paper-2013</a> |
        <a
          href="http://watson.nci.nih.gov/cran_mirror/web/packages/RCircos/index.html"
          >web</a
        >
        ]
      </li>
      <li>
        <strong
          ><a href="https://github.com/mkanai/fujiplot">fujiplot</a></strong
        >
        - A circos representation of multiple GWAS results. [
        <a href="https://www.nature.com/articles/s41588-018-0047-6"
          >paper-2018</a
        >
        ]
      </li>
    </ul>
    <h2 id="database-access">Database Access</h2>
    <ul>
      <li>
        <a href="http://www.ncbi.nlm.nih.gov/books/NBK179288/"
          >Entrez Direct: E-utilities on the UNIX command line</a
        >
        - UNIX command line tools to access NCBI’s databases programmatically.
        Instructions to install and examples are found in the link.
      </li>
    </ul>
    <h2 id="resources">Resources</h2>
    <h3 id="becoming-a-bioinformatician">Becoming a Bioinformatician</h3>
    <ul>
      <li>
        <a href="http://blog.fejes.ca/?p=2418">What is a bioinformatician</a>
      </li>
      <li>
        <a
          href="http://www.ploscompbiol.org/article/info:doi%2F10.1371%2Fjournal.pcbi.1003496"
          >Bioinformatics Curriculum Guidelines: Toward a Definition of Core
          Competencies</a
        >
      </li>
      <li>
        <a
          href="http://caseybergman.wordpress.com/2012/07/31/top-n-reasons-to-do-a-ph-d-or-post-doc-in-bioinformaticscomputational-biology/"
          >Top N Reasons To Do A Ph.D. or Post-Doc in
          Bioinformatics/Computational Biology</a
        >
      </li>
      <li>
        <a
          href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-1-104"
          >A 10-Step Guide to Party Conversation For Bioinformaticians</a
        >
        - Here is a step-by-step guide on how to convey concepts to people not
        involved in the field when asked the question: ‘So, what do you do?’
      </li>
      <li>
        <a href="https://www.youtube.com/watch?v=uwsjwMO-TEA"
          >A History Of Bioinformatics (In The Year 2039)</a
        >
        - A talk by C. Titus Brown on his take of looking back at bioinformatics
        from the year 2039. His notes for this talk can be found
        <a href="http://ivory.idyll.org/blog/2014-bosc-keynote.html">here</a>.
      </li>
      <li>
        <a
          href="http://madhadron.com/posts/2012-03-26-a-farewell-to-bioinformatics.html"
          >A farewell to bioinformatics</a
        >
        - A critical view of the state of bioinformatics.
      </li>
      <li>
        <a
          href="http://www.acgt.me/blog/2014/3/25/101-questions-a-new-series-of-interviews-with-notable-bioinformaticians"
          >A Series of Interviews with Notable Bioinformaticians</a
        >
        - Dr. Keith Bradnam “thought it might be instructive to ask a simple
        series of questions to a bunch of notable bioinformaticians to assess
        their feelings on the current state of bioinformatics research, and
        maybe get any tips they have about what has been useful to their
        bioinformatics careers.”
      </li>
      <li>
        <a href="https://github.com/ossu/bioinformatics"
          >Open Source Society University on Bioinformatics</a
        >
        - Solid path for those of you who want to complete a Bioinformatics
        course on your own time, for free, with courses from the best
        universities in the World.
      </li>
      <li>
        <a href="http://rosalind.info/">Rosalind</a> - Rosalind is a platform
        for learning bioinformatics through problem solving.
      </li>
      <li>
        <a
          href="http://www.opiniomics.org/a-guide-for-the-lonely-bioinformatician/"
          >A guide for the lonely bioinformatician</a
        >
        - This guide is aimed at bioinformaticians, and is meant to guide them
        towards better career development.
      </li>
      <li>
        <a href="https://doi.org/10.1093/bib/bby063"
          >A brief history of bioinformatics</a
        >
      </li>
    </ul>
    <h3 id="bioinformatics-on-github">Bioinformatics on GitHub</h3>
    <ul>
      <li>
        <a href="https://github.com/HussainAther/awesome-alternative-splicing"
          >Awesome-alternative-splicing</a
        >
        - List of resources on alternative splicing including software,
        databases, and other tools..
      </li>
    </ul>
    <h3 id="sequencing">Sequencing</h3>
    <ul>
      <li>
        <a href="https://youtu.be/6Is3W7JkFp8"
          >Next-Generation Sequencing Technologies - Elaine Mardis (2014)</a
        >
        [1:34:35] - Excellent (technical) overview of next-generation and
        third-generation sequencing technologies, along with some applications
        in cancer research.
      </li>
      <li>
        <a href="https://liorpachter.wordpress.com/seq/"
          >Annotated bibliography of *Seq assays</a
        >
        - List of ~100 papers on various sequencing technologies and assays
        ranging from transcription to transposable element discovery.
      </li>
      <li>
        <a
          href="http://www.illumina.com/content/dam/illumina-marketing/documents/applications/ngs-library-prep/ForAllYouSeqMethods.pdf"
          >For all you seq… (PDF)</a
        >
        (3456x5471) - Massive infographic by Illumina on illustrating how many
        sequencing techniques work. Techniques cover protein-protein
        interactions, RNA transcription, RNA-protein interactions, RNA low-level
        detection, RNA modifications, RNA structure, DNA rearrangements and
        markers, DNA low-level detection, epigenetics, and DNA-protein
        interactions. References included.
      </li>
    </ul>
    <h3 id="rna-seq">RNA-Seq</h3>
    <ul>
      <li>
        <a href="https://www.biostars.org/p/52152/"
          >Review papers on RNA-seq (Biostars)</a
        >
        - Includes lots of seminal papers on RNA-seq and analysis methods.
      </li>
      <li>
        <a href="https://github.com/griffithlab/rnaseq_tutorial"
          >Informatics for RNA-seq: A web resource for analysis on the cloud</a
        >
        - Educational resource on performing RNA-seq analysis in the cloud using
        Amazon AWS cloud services. Topics include preparing the data,
        preprocessing, differential expression, isoform discovery, data
        visualization, and interpretation.
      </li>
      <li>
        <a href="http://rnaseq.uoregon.edu/">RNA-seqlopedia</a> - RNA-seqlopedia
        provides an awesome overview of RNA-seq and of the choices necessary to
        carry out a successful RNA-seq experiment.
      </li>
      <li>
        <a
          href="http://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0881-8"
          >A survey of best practices for RNA-seq data analysis</a
        >
        - Gives awesome roadmap for RNA-seq computational analyses, including
        challenges/obstacles and things to look out for, but also how you might
        integrate RNA-seq data with other data types.
      </li>
      <li>
        <a href="https://www.youtube.com/watch?v=5NiFibnbE8o"
          >Stories from the Supplement</a
        >
        [46:39] - Dr. Lior Pachter shares his stories from the supplement for
        well-known RNA-seq analysis software CuffDiff and
        <a href="http://cole-trapnell-lab.github.io/cufflinks/">Cufflinks</a>
        and explains some of their methodologies.
      </li>
      <li>
        <a
          href="https://en.wikipedia.org/wiki/List_of_RNA-Seq_bioinformatics_tools"
          >List of RNA-seq Bioinformatics Tools</a
        >
        - Extensive list on Wikipedia of RNA-seq bioinformatics tools needed in
        analysis, ranging from all parts of an analysis pipeline from quality
        control, alignment, splice analysis, and visualizations.
      </li>
      <li>
        <a href="https://github.com/crazyhottommy/RNA-seq-analysis"
          >RNA-seq Analysis</a
        >
        -
        <span class="citation" data-cites="crazyhottommy">[@crazyhottommy]</span
        >(https://github.com/crazyhottommy)’s notes on various steps and
        considerations when doing RNA-seq analysis.
      </li>
    </ul>
    <h3 id="chip-seq">ChIP-Seq</h3>
    <ul>
      <li>
        <a href="https://github.com/crazyhottommy/ChIP-seq-analysis"
          >ChIP-seq analysis notes from Tommy Tang</a
        >
        - Resources on ChIP-seq data which include papers, methods, links to
        software, and analysis.
      </li>
    </ul>
    <h3 id="youtube-channels-and-playlists">YouTube Channels and Playlists</h3>
    <ul>
      <li>
        <a
          href="https://www.genome.gov/12514288/current-topics-in-genome-analysis-2016-course-syllabus-handouts-and-videos/"
          >Current Topics in Genome Analysis 2016</a
        >
        - Excellent series of fourteen lectures given at NIH about current
        topics in genomics ranging from sequence analysis, to sequencing
        technologies, and even more translational topics such as genomic
        medicine.
      </li>
      <li>
        <a href="https://www.youtube.com/user/GenomeTV">GenomeTV</a> - “GenomeTV
        is NHGRI’s collection of official video resources from lectures, to news
        documentaries, to full video collections of meetings that tackle the
        research, issues and clinical applications of genomic research.”
      </li>
      <li>
        <a href="https://www.youtube.com/user/LeadingStrand">Leading Strand</a>
        - Keynote lectures from Cold Spring Harbor Laboratory (CSHL) Meetings.
        More on
        <a href="http://theleadingstrand.cshl.edu/">The Leading Strand</a>.
      </li>
      <li>
        <a
          href="https://www.youtube.com/playlist?list=PLqLDR0CTP9_pboZCk6gR9Zn4kW7h9XWJI"
          >Genomics, Big Data and Medicine Seminar Series</a
        >
        - “Our seminars are dedicated to the critical intersection of GBM,
        delving into ‘bleeding edge’ technology and approaches that will deeply
        shape the future.”
      </li>
      <li>
        <a href="https://www.youtube.com/user/RafalabChannel/videos"
          >Rafael Irizarry’s Channel</a
        >
        - Dr. Rafael Irizarry’s lectures and academic talks on statistics for
        genomics.
      </li>
      <li>
        <a href="https://www.youtube.com/user/nihvcast"
          >NIH VideoCasting and Podcasting</a
        >
        - “NIH VideoCast broadcasts seminars, conferences and meetings live to a
        world-wide audience over the Internet as a real-time streaming video.”
        Not exclusively genomics and bioinformatics video but many great talks
        on domain specific use of bioinformatics and genomics.
      </li>
    </ul>
    <h3 id="blogs">Blogs</h3>
    <ul>
      <li>
        <a href="http://www.acgt.me/">ACGT</a> - Dr. Keith Bradnam writes about
        this “thoughts on biology, genomics, and the ongoing threat to humanity
        from the bogus use of bioinformatics acroynums.”
      </li>
      <li>
        <a href="http://www.opiniomics.org/">Opiniomics</a> - Dr. Mick Watson
        write on bioinformatics, genomes, and biology.
      </li>
      <li>
        <a href="https://liorpachter.wordpress.com/">Bits of DNA</a> - Dr. Lior
        Pachter writes review and commentary on computational biology.
      </li>
      <li>
        <a href="http://www.michaeleisen.org/blog/">it is NOT junk</a> -
        Dr. Michael Eisen writes “a blog about genomes, DNA, evolution, open
        science, baseball and other important things”
      </li>
    </ul>
    <h3 id="miscellaneous">Miscellaneous</h3>
    <ul>
      <li>
        <a href="https://github.com/jtleek/genomicspapers/"
          >The Leek group guide to genomics papers</a
        >
        - Expertly curated genomics papers to get up to speed on genomics,
        RNA-seq, statistics (used in genomics), software development, and more.
      </li>
      <li>
        <a href="https://doi.org/10.1371/journal.pcbi.1003662"
          >A New Online Computational Biology Curriculum</a
        >
        - “This article introduces a catalog of several hundred free video
        courses of potential interest to those wishing to expand their knowledge
        of bioinformatics and computational biology. The courses are organized
        into eleven subject areas modeled on university departments and are
        accompanied by commentary and career advice.”
      </li>
      <li>
        <a href="http://www.foo.be/docs/tpj/issues/vol1_2/tpj0102-0001.html"
          >How Perl Saved the Human Genome Project</a
        >
        - An anecdote by Lincoln D. Stein on the importance of the Perl
        programming language in the Human Genome Project.
      </li>
      <li>
        <a
          href="https://liacs.leidenuniv.nl/~hoogeboomhj/mcb/nature_primer.html"
          >Educational Papers from Nature Biotechnology and PLoS Computational
          Biology</a
        >
        - Page of links to primers and short educational articles on various
        methods used in computational biology and bioinformatics.
      </li>
      <li>
        <a href="https://peerj.com/collections/45-bioinformatics-software/"
          >The PeerJ Bioinformatics Software Tools Collection</a
        >
        - Collection of tools curated by Keith Crandall and Claus White, aimed
        at collating the most interesting, innovative, and relevant
        bioinformatics tools articles in PeerJ.
      </li>
    </ul>
    <h2 id="online-networking-groups">Online networking groups</h2>
    <ul>
      <li>
        <a href="https://discord.com/invite/3uxbPns"
          >Bioinformatics (on Discord)</a
        >
        - a Discord server for general bioinformatics
      </li>
      <li>
        <a
          href="https://www.reddit.com/r/bioinformatics/comments/7ndwm1/rbioinformatics_slack_channel_and_an_open_call/"
          >r-bioinformatics</a
        >
        - the official Slack workspace of r/bioinformatics (<a
          href="https://www.reddit.com/message/compose/?to=apfejes&amp;subject=Request%20to%20join%20the%20r/bioinformatics%20Slack%20group&amp;message=I%20would%20like%20to%20request%20to%20join%20the%20r/bioinformatics%20Slack%20group"
          >send a direct message to apfejes on reddit</a
        >)
      </li>
      <li>
        <a href="https://bioinformaticsgrx.es/">BioinformaticsGRX</a> - A
        community of bioinformaticians based in Granada, Spain
      </li>
      <li>
        <a href="https://comunidadbioinfo.github.io/"
          >Comunidad de Desarolladores de Software en Bioinformática</a
        >
        - A community of bioinformaticians centered in Latin America
      </li>
      <li>
        <a href="https://combine.org.au/">COMBINE</a> - An Austrialian group for
        bioinformatics students
      </li>
    </ul>
    <h2 id="license">License</h2>
    <p>
      <a href="https://creativecommons.org/publicdomain/zero/1.0/"
        ><img
          src="http://mirrors.creativecommons.org/presskit/buttons/88x31/svg/cc-zero.svg"
          alt="CC0"
      /></a>
    </p>
  </body>
</html>
